### PR TITLE
api: stream changes to `Announcement`s to clients using SSE

### PR DESCRIPTION
**Pull request by colourdelete**
Note: not using Protocol Buffers and gRPC as it isn't ready yet.

`GET /api/announcements/changes`
```
data: <serialized changed Announcement>
```

### CODE REVIEW

**Review by ApocalypseCalculator — APPROVED** [review]

File: core/api/views/stream.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        self.signal = signal
+        self.serializer = serializer
+        self.q = LifoQueue()
+        self.__setup()
+
+    def __setup(self):
+        self.signal.connect(self.__receive, sender=self.model)
+
+    def __del__(self):
+        self.signal.disconnect(self.__receive, sender=self.model)
```
Comment (colourdelete) [inline core/api/views/stream.py:28]: this should run (be GC'd) when the connection is GC'd

**Review by yaxollum — APPROVED** [review]
LGTM (I don't know much about signals and streams though)

File: core/api/views/stream.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    def __receive(self, instance, **_):
+        self.q.put(instance)
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        instance = self.q.get()
+        return f"data: {json.dumps(self.serializer(instance).data)}\n"
```
Comment (hillcrestpaul0719) [inline core/api/views/stream.py:38]: It might be easier to parse without the `data: ` prefix

**Review by hillcrestpaul0719 — APPROVED** [review]
I don't know much about Streaming requests but otherwise lgtm
commented some thoughts
sorry for delayed review :(

File: core/api/views/announcement.py [hunk 3]
```diff
@@ -1,10 +1,16 @@
-from django.db.models import Q
+import json
+from typing import List
+from weakref import WeakValueDictionary
```
[the remark this answers is not in the document]
Comment (colourdelete) [reply core/api/views/announcement.py:3]: nope, will fix in like 1 min, thanks!

File: core/api/views/stream.py (hunk 2 above)
Comment (colourdelete) [reply core/api/views/stream.py:37]: its required i belive for SSE: https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events#event_stream_format

Comment (colourdelete) [reply core/api/views/stream.py:37]: oh wait, should I set this as a specific event?

**Pull request merged**